opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -461,7 +461,9 @@ message HistogramDataPoint {
   // The sum of the bucket_counts must equal the value in the count field.
   //
   // The number of elements in bucket_counts array must be by one greater than
-  // the number of elements in explicit_bounds array.
+  // the number of elements in explicit_bounds array. The exception to this rule
+  // is when the length of bucket_counts is 0, then the length of explicit_bounds
+  // must also be 0.
   repeated fixed64 bucket_counts = 6;
```

Comment:
If the `bucket_counts` is zero, then all the other fields also have to be zero / omitted, else the data would be contradictory.

I suppose this isn't necessarily invalid, but its definitely an edge case.

Reply:
The data model [says](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#histogram)

> Bucket counts are optional. A Histogram without buckets conveys a population in terms of only the sum and count, and may be interpreted as a histogram with single bucket covering (-Inf, +Inf).

So the rest of the fields need not be zero in this case

Reply:
Is it worth copying that phrase about bucket counts being optional from the data model spec to the comment in the proto file?

Reply:
Ah I forgot that there is a separate top level `count` field. Ok so there's still useful data that can be transmitted without bucket counts.